icechunk-python/src/errors.rs

```diff
@@ -77,24 +82,32 @@ impl From<SessionError> for PyIcechunkStoreError {
 impl From<PyIcechunkStoreError> for PyErr {
     fn from(error: PyIcechunkStoreError) -> Self {
         match error {
-            PyIcechunkStoreError::SessionError(SessionError::Conflict {
-                expected_parent,
-                actual_parent,
+            PyIcechunkStoreError::SessionError(SessionError {
+                kind: SessionErrorKind::Conflict { expected_parent, actual_parent },
+                ..
             }) => PyConflictError::new_err(PyConflictErrorData {
                 expected_parent: expected_parent.map(|s| s.to_string()),
                 actual_parent: actual_parent.map(|s| s.to_string()),
             }),
-            PyIcechunkStoreError::SessionError(SessionError::RebaseFailed {
-                snapshot,
-                conflicts,
+            PyIcechunkStoreError::SessionError(SessionError {
+                kind: SessionErrorKind::RebaseFailed { snapshot, conflicts },
+                ..
             }) => PyRebaseFailedError::new_err(PyRebaseFailedData {
                 snapshot: snapshot.to_string(),
                 conflicts: conflicts.iter().map(PyConflict::from).collect(),
             }),
             PyIcechunkStoreError::PyKeyError(e) => PyKeyError::new_err(e),
             PyIcechunkStoreError::PyValueError(e) => PyValueError::new_err(e),
             PyIcechunkStoreError::PyError(err) => err,
-            _ => IcechunkError::new_err(error.to_string()),
+            error => {
```

Comment:
this generates a message for the python exception that captures the full list of "causes" and the full span captured by the error